platform-dom/src/main/java/com/softicar/platform/dom/elements/number/decimal/DomDecimalInput.java

```diff
@@ -0,0 +1,86 @@
+package com.softicar.platform.dom.elements.number.decimal;
+
+import com.softicar.platform.common.core.locale.CurrentLocale;
+import com.softicar.platform.common.core.number.BigDecimalMapper;
+import com.softicar.platform.common.core.number.formatter.BigDecimalFormatter;
+import com.softicar.platform.common.core.number.formatter.BigDecimalScaleApplier;
+import com.softicar.platform.common.core.number.parser.BigDecimalParser;
+import com.softicar.platform.dom.DomI18n;
+import com.softicar.platform.dom.elements.input.AbstractDomNumberInput;
+import com.softicar.platform.dom.input.DomInputException;
+import com.softicar.platform.dom.input.DomTextInput;
+import java.math.BigDecimal;
+
+/**
+ * A {@link DomTextInput} for decimal values that formats and parses values
+ * according to {@link CurrentLocale}.
+ *
+ * @author Oliver Richers
+ */
+public class DomDecimalInput<T extends Number> extends AbstractDomNumberInput<T> {
+
+	private final BigDecimalMapper<T> mapper;
+	private int scale = -1;
```

Comment:
I think this is confusing, and that we should rather use a `null` `Integer` to indicate an undefined scale.
After all, even though rarely used, negative scales are a thing, in a `BigDecimal` context. Yet, the value of `-1` here is not actually meant to be a negative scale - it's merely used to not have to bother with `null` checks.

Reply:
changed this, even allowing negative scale, although it will have no effect